core/src/Revolution/Processors/Workspace/Lexicon/ReloadFromBase.php

```diff
@@ -41,42 +42,53 @@ public function getLanguageTopics()
      */
     public function process()
     {
-        $language = 'en';
-        $namespace = 'core';
-        $corePath = $this->modx->getOption('core_path', null, MODX_CORE_PATH);
+        $namespace = $this->getProperty('namespace');
+        $topic = $this->getProperty('lexiconTopic');
+        $language = $this->getProperty('language');
+
+        if (empty($namespace) || empty($topic) || empty($language)) {
+            $this->modx->log(modX::LOG_LEVEL_WARN, $this->modx->lexicon('lexicon_revert_error'));
+            $this->modx->log(modX::LOG_LEVEL_INFO, 'COMPLETED');
+
+            return $this->success();
+        }
+
+        $namespaceObj = $this->modx->getObject(modNamespace::class, $namespace);
+        $namespacePath = $namespaceObj->getCorePath();
```

Comment:
A null exception can occur here if the namespace doesn't exist or isn't specified. That's definitely an edge case but I'd prefer that to be caught by checking if $namespaceObj is in fact a modNamespace.

Reply:
You're checking if the namespace is empty but not if it is a valid namespace that exists and can be loaded. So if an invalid namespace is passed, `$this->modx->getObject()` will return null, and a fatal error will occur trying to call `getCorePath()` on null.